### ISSUE TITLE
cannot use group norm in torchvision API

### ISSUE THREAD

**Issue by KaiHoo** [pytorch/pytorch#34042]
## 🐛 Bug

Cannot use nn.GroupNorm for "norm_layer" in torchvision.models.resnet50. However, nn.GroupNorm is supposed to be OK since there are codes to initialize the weights of nn.GroupNorm in the resnet50 file. 

## To Reproduce

>>> import torch.nn as nn
>>> import torchvision
>>> net = torchvision.models.resnet50(norm_layer=nn.BatchNorm2d)
>>> net = torchvision.models.resnet50(norm_layer=nn.GroupNorm)

## Expected behavior

Both nets should be OK, however the second raised an error in torchvision/models/resnet.py line 143

## Environment

PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.13.6
GCC version: Could not collect
CMake version: version 3.14.5

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip3] numpy==1.18.1
[pip3] torch==1.4.0
[pip3] torchvision==0.5.0
[conda] torch                     1.4.0                    pypi_0    pypi
[conda] torchvision               0.5.0                    pypi_0    pypi
